src/FsAutoComplete/LspServers/AdaptiveFSharpLspServer.fs

```diff
@@ -814,28 +937,39 @@ type AdaptiveFSharpLspServer(workspaceLoader: IWorkspaceLoader, lspClient: FShar
           >> Log.addContextDestructured "file" file.Lines.FileName
         )
 
+        Async.Start(
+          async {
+            let checkErrors = parseAndCheck.GetParseResults.Diagnostics
+            let parseErrors = parseAndCheck.GetCheckResults.Diagnostics
 
-        let checkErrors = parseAndCheck.GetParseResults.Diagnostics
-        let parseErrors = parseAndCheck.GetCheckResults.Diagnostics
+            let errors =
+              Array.append checkErrors parseErrors
+              |> Array.distinctBy (fun e ->
+                e.Severity, e.ErrorNumber, e.StartLine, e.StartColumn, e.EndLine, e.EndColumn, e.Message)
+
+            notifications.Trigger(NotificationEvent.ParseError(errors, file.Lines.FileName), ct)
+          },
+          ct
+        )
 
-        let errors =
-          Array.append checkErrors parseErrors
-          |> Array.distinctBy (fun e ->
-            e.Severity, e.ErrorNumber, e.StartLine, e.StartColumn, e.EndLine, e.EndColumn, e.Message)
 
-        NotificationEvent.ParseError(errors, file.Lines.FileName)
-        |> notifications.Trigger
 
 
-        do! analyzeFile config (file.Lines.FileName, file.Version, file.Lines, parseAndCheck)
+        Async.Start(analyzeFile config (file.Lines.FileName, file.Version, file.Lines, parseAndCheck), ct)
 
-        // LargeObjectHeap gets fragmented easily for really large files, which F# can easily have.
-        // Yes this seems excessive doing this every time we type check but it's the best current kludge.
-        System.Runtime.GCSettings.LargeObjectHeapCompactionMode <-
-          System.Runtime.GCLargeObjectHeapCompactionMode.CompactOnce
 
-        GC.Collect()
-        GC.WaitForPendingFinalizers()
+        Async.Start(
```

Comment:
Moving things to Async calls to allow type checks to come back faster. Not as strictly necessary anymore given our new method of completions.